### DISCUSSION TITLE
Github Action을 통한 SwiftLint적용

### DISCUSSION THREAD

**Discussion by HanGyeongjun** [Team-ShortcutsZip/MacC-Team-HappyAnding#384] [General]
Sprint6에서 코드 리팩토링이 어느 정도 진행되어서 Sprint7/Sptint8 부터는 Github Action을 통한 SwiftLint를 적용시켜봐도 좋겠다는 생각이 들었습니다!
간단히 조사를 해 본 결과 충분히 ShortcutsZip에 적용시킬 수 있다고 판단해서 공유합니다.

- 간단하게 적용시킬 수 있는가?
    - 현재 Github 마켓플레이스에 [SwiftLint를 위한 Action이 존재](https://github.com/marketplace/actions/github-action-for-swiftlint)해서 따로 제작할 필요가 없이 바로 프로젝트에 적용할 수 있습니다.
    - 어떤 룰을 적용할지에 대한 논의는 필요합니다.
    - 최초 설정만 해 주면 기존과 똑같이 프로젝트 사용이 가능하고, 큰 학습 비용이 발생하지 않습니다.
    
- 언제/어떻게  작동하는 Action인가?
    - develop 브렌치에 Pull Request가 발생했을 시 작동합니다.
    - Action이 코드를 검사해 Lint에 맞지 않는 부분이 있으면 경고를 표시하고, 통과 시 댓글로 알려줍니다.
    
- 왜 Lint를 프로젝트 말고 Github Action에 적용시키나?
    - ShortcutsZip이 커졌기 때문에 프로젝트에 적용시키면 Lint로 인한 경고가 많이 발생하므로 비효율적이라고 생각합니다.
    - Action에 적용시켜 PR시 일괄적으로 확인하고 크리티컬한 부분을 수정하는 과정이 더 효율적일 것 같습니다!
    
- 이 Action을 적용하면 어떤 이점이 있나?
    - Lint 자동화 -> 깔-끔한 코드
    - 간단한 컨벤션 관련 코드리뷰 자동화
    - 추가적인 Action적용을 위한 학습 가능

**Comment by HanGyeongjun** [comment]
관련 블로그 링크입니다!
https://velog.io/@hyesuuou/SwiftLint-Github-Action%EC%9C%BC%EB%A1%9C-%EC%B2%B4%ED%81%AC%ED%95%98%EA%B8%B0
https://didu-story.tistory.com/385?category=1022250

두 번째 블로그는 아카데미 요모조모 팀원분의 블로그로, 해당 레포에 액션이 적용되어있어서 참고해보면 좋을 것 같습니다.
[요모조모 Repo](https://github.com/DeveloperAcademy-POSTECH/MacC-Team-EarthValley80)

**Comment by JIWON1923** [comment]
룰은 저희 코드 컨벤션 기반으로 작성해도 좋을 것 같네요

컨벤션 부분은 SwiftLint를 통해 확인하면 가독성도 좋아지지만, 코드리뷰하기도 수월할 것 같아서 찬성합니다!
(git Action으로 하는 것은 좋지만, 깔끔한 코드 일관성 있는 코드를 위해서라면 오류를 감수하고 대공사 한 번 하는게 좋긴 할 것 같아요)
